pkg/kubectl/cmd/config/create_cluster.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"errors"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+)
+
+type createClusterOptions struct {
+	pathOptions           *pathOptions
+	name                  string
+	server                string
+	apiVersion            string
+	insecureSkipTLSVerify bool
+	certificateAuthority  string
+}
+
+func NewCmdConfigSetCluster(out io.Writer, pathOptions *pathOptions) *cobra.Command {
+	options := &createClusterOptions{pathOptions: pathOptions}
+
+	cmd := &cobra.Command{
+		Use:   "set-cluster name [server] [insecure-skip-tls-verify] [certificate-authority] [api-version]",
```

Comment:
I'm not a huge fan of set-user-name as a command.  "login" is more common, or even "set-credentials".  What does gcloud use?


> On Jan 8, 2015, at 7:54 PM, Jeff Lowdermilk <notifications@github.com> wrote:
> 
> In pkg/kubectl/cmd/config/create_cluster.go:
> 
> > +)
> > +
> > +type createClusterOptions struct {
> > +	pathOptions           *pathOptions
> > +	name                  string
> > +	server                string
> > +	apiVersion            string
> > +	insecureSkipTLSVerify bool
> > +	certificateAuthority  string
> > +}
> > +
> > +func NewCmdConfigSetCluster(out io.Writer, pathOptions *pathOptions) *cobra.Command {
> > +	options := &createClusterOptions{pathOptions: pathOptions}
> > +
> > +	cmd := &cobra.Command{
> > +		Use:   "set-cluster name [server] [insecure-skip-tls-verify] [certificate-authority] [api-version]",
> Ditto, re flags for optional args. Added benefit of having cobra parse the type and set a default, so you can do e.g.
> 
> cmd.Flags().Bool("insecure-skip-tls-verify", false, "If true, the server's certificate will not be checked for validity.") 
> —
> Reply to this email directly or view it on GitHub.
>